### DISCUSSION TITLE
ChromiumWebBrowse rendering issue on WPF application.

### DISCUSSION THREAD

**Discussion by karthikcefsharp** [cefsharp/CefSharp#4319] [General]
Hello Team,

We are working on the WPF application on .NET 4.8 with CefSharp.Common, CefSharp.WPF, cef.redist.x86 libraries.
In that we are using the ChromiumWebBrowser to render our internal web portal into our WPF application.
Recently we have upgraded from 75.1.142 to the latest 105.3.33 version. After that the ChromiumWebBrowser is not rendering the web response properly in some client systems.
Mostly we noticed problems from WIN11 (not only) machines.

Here I attached the screenshots for the various rendering issues and the scenarios. 

1. Not able to see the page but the elements are loaded in the background of browser. (Can able to see, based on the cursor movement in the screen).
![ChromiumBrowserRenderIssue_2](https://user-images.githubusercontent.com/119283820/204226916-58fa5cc6-00d9-4d87-9a9d-b7c90a0a4b2f.PNG)
![ChromiumBrowserRenderIssue_3](https://user-images.githubusercontent.com/119283820/204226949-4abda975-8fce-4ba0-81d4-bc33d5cf2580.PNG)

2. Partially elements are loaded in the browser.
![ChromiumBrowserRenderIssue_1](https://user-images.githubusercontent.com/119283820/204226984-bdb4c48c-a39c-4bb1-8374-6d77c14766b4.PNG)

3. After sometime (later an hour), we can see the full page in the browser (very rare scenario).

We tried to upgrade 105.3.33 to latest version 107.1.90.0, but same result. Also we can see some new errors found in the debug.log.

[1125/152032.795:ERROR:viz_main_impl.cc(186)] Exiting GPU process due to errors during initialization
[1125/152032.928:ERROR:gpu_init.cc(537)] Passthrough is not supported, GL is disabled, ANGLE is 
[1125/152544.858:ERROR:angle_platform_impl.cc(43)] Display.cpp:1004 (initialize): ANGLE Display::initialize error 0: Internal Vulkan error (-3): Initialization of an object could not be completed for implementation-specific reasons, in ..\..\third_party\angle\src\libANGLE\renderer\vulkan\RendererVk.cpp, initialize:1398.
[1125/152544.858:ERROR:gl_display.cc(508)] EGL Driver message (Critical) eglInitialize: Internal Vulkan error (-3): Initialization of an object could not be completed for implementation-specific reasons, in ..\..\third_party\angle\src\libANGLE\renderer\vulkan\RendererVk.cpp, initialize:1398.
[1125/152544.858:ERROR:gl_display.cc(920)] eglInitialize SwANGLE failed with error EGL_NOT_INITIALIZED

Here I also attached the debug.log file from CefSharp.
[debug.log](https://github.com/cefsharp/CefSharp/files/10101725/debug.log)

Below are the coding details.

```
public void InitCefSharp()
{
            //Monitor parent process exit and close subprocesses if parent process exits first
            CefSharpSettings.SubprocessExitIfParentProcessClosed = true;
            CefSharpSettings.ShutdownOnExit = false;

            CefSettings cefSettings = new CefSettings
            {
                CachePath = System.IO.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData),
                    @"Tech Tool\BrowserCache"),
                PersistSessionCookies = true,
                BrowserSubprocessPath = System.IO.Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"BrowserDependencies\CefSharp.BrowserSubprocess.exe"),
                LocalesDirPath = System.IO.Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"BrowserDependencies\"),
                ResourcesDirPath = System.IO.Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"BrowserDependencies\")
            };

            CefSharp.Cef.Initialize(cefSettings);
}

// For Initializing.
this.browser = new ChromiumWebBrowser(uri);

// For Updating.
this.browser.Load(uri);
this.browser.Reload();
```
Could you please assist on this issue.

Regards.

**Comment by amaitland** [comment]
In future please use [Discussions](https://github.com/cefsharp/CefSharp/discussions) to ask support related  questions.

> Exiting GPU process due to errors during initialization

Your application needs an [app.manifest](https://github.com/cefsharp/CefSharp/wiki/Quick-Start-For-MS-.Net-4.x#40-add-appmanifest-to-your-application) with compatibility entries from GPU detection to work correctly. 

There's an example at https://github.com/cefsharp/CefSharp.MinimalExample/blob/master/CefSharp.MinimalExample.Wpf/app.manifest

For testing purposes you can use https://github.com/cefsharp/CefSharp.MinimalExample which is kept up to date. 

For any machine that has an Intel GPU make sure it's running the latest drivers.
